pkg/distributor/distributor.go

```diff
@@ -765,6 +770,30 @@ func (d *Distributor) checkIngestLimit(tenantID string, req *distributormodel.Pu
 	return nil
 }
 
+func (d *Distributor) checkUsageGroupsIngestLimit(tenantID string, groupsInRequest []string, req *distributormodel.PushRequest) error {
+	l := d.limits.IngestionLimit(tenantID)
+	if l == nil || len(l.UsageGroups) == 0 {
+		return nil
+	}
+
+	for _, group := range groupsInRequest {
+		if limit, ok := l.UsageGroups[group]; ok {
+			if limit.LimitReached {
+				if d.ingestionLimitsSampler.AllowRequest(tenantID, l.Sampling) {
+					return nil
+				}
+				limitResetTime := time.Unix(l.LimitResetTime, 0).UTC().Format(time.RFC3339)
+				validation.DiscardedProfiles.WithLabelValues(string(validation.IngestLimitReached), tenantID).Add(float64(req.TotalProfiles))
+				validation.DiscardedBytes.WithLabelValues(string(validation.IngestLimitReached), tenantID).Add(float64(req.TotalBytesUncompressed))
+				return connect.NewError(connect.CodeResourceExhausted,
+					fmt.Errorf("limit of %s/%s reached for usage group %s, next reset at %s", humanize.IBytes(uint64(limit.PeriodLimitMb*1024*1024)), l.PeriodType, group, limitResetTime))
+			}
+		}
+	}
```

Comment:
Very slight nit, but I think inverting some of these if statements will help readability (please triple check I inverted these correctly, it's hard to edit code in GitHub).

```suggestion
	for _, group := range groupsInRequest {
		limit, ok := l.UsageGroups[group]
		if !ok || !limit.LimitReached {
			return continue
		}

		if d.ingestionLimitsSampler.AllowRequest(tenantID, l.Sampling) {
			return nil
		}

		limitResetTime := time.Unix(l.LimitResetTime, 0).UTC().Format(time.RFC3339)
		validation.DiscardedProfiles.WithLabelValues(string(validation.IngestLimitReached), tenantID).Add(float64(req.TotalProfiles))
		validation.DiscardedBytes.WithLabelValues(string(validation.IngestLimitReached), tenantID).Add(float64(req.TotalBytesUncompressed))
		return connect.NewError(connect.CodeResourceExhausted,
			fmt.Errorf("limit of %s/%s reached for usage group %s, next reset at %s", humanize.IBytes(uint64(limit.PeriodLimitMb*1024*1024)), l.PeriodType, group, limitResetTime))
	}
```

Reply:
Great suggestion! Applied manually in a0ec5558d.